diff --git a/node_modules/@shopify/shopify-app-remix/dist/esm/react/components/AppProvider/AppProvider.mjs b/node_modules/@shopify/shopify-app-remix/dist/esm/react/components/AppProvider/AppProvider.mjs
index 61dc2ee..0c69258 100644
--- a/node_modules/@shopify/shopify-app-remix/dist/esm/react/components/AppProvider/AppProvider.mjs
+++ b/node_modules/@shopify/shopify-app-remix/dist/esm/react/components/AppProvider/AppProvider.mjs
@@ -1,6 +1,6 @@
 import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
 import { AppProvider as AppProvider$1 } from '@shopify/polaris';
-import englishI18n from '@shopify/polaris/locales/en.json' with { type: 'json' };
+import englishI18n from '@shopify/polaris/locales/en.json';
 import { APP_BRIDGE_URL } from '../../const.mjs';
 import { RemixPolarisLink } from '../RemixPolarisLink.mjs';
 
